docs/NEW_LANGUAGE_FEATURE_INTEGRATION_PROCESS.md

```diff
@@ -0,0 +1,25 @@
+# New Language Feature Check Integration Process [WIP]
+
+## Check Update Procedure
+
+### Consider Similar Tokens
+### Java Enhancement Proposals (JEPS)
+### Always Impacted Checks
+
+- illegalToken
+- illegalTokenText 
+- whitespace
+- indentation
```

Comment:
Use the correct casing that is in our docs, and hyperlink it, like this: [AnnotationLocation](https://checkstyle.sourceforge.io/checks/annotation/annotationlocation.html#AnnotationLocation)